curator-framework/src/test/java/org/apache/curator/framework/imps/TestFramework.java

```diff
@@ -1073,7 +1080,41 @@ public void testCreateModes() throws Exception
         }
         finally
         {
-            CloseableUtils.closeQuietly(client);
+        	CloseableUtils.closeQuietly(client);
+        }
+    }
+    
+    @Test
+    public void testConfigurableZookeeper() throws Exception
+    {
+    	CuratorFramework client = null;
+        try
+        {
+        	ZKClientConfig zkClientConfig = new ZKClientConfig();
+    		String zookeeperRequestTimeout = "30000";
+    		zkClientConfig.setProperty(ZKClientConfig.ZOOKEEPER_REQUEST_TIMEOUT, zookeeperRequestTimeout);
+    		client = CuratorFrameworkFactory.newClient(server.getConnectString(), 30000, 30000, new RetryOneTime(1), zkClientConfig);
+            client.start();
+        	
+            byte[] writtenBytes = {1, 2, 3};
+            client.create().forPath("/test", writtenBytes);
+
+            byte[] readBytes = client.getData().forPath("/test");
+            assertArrayEquals(writtenBytes, readBytes);
+            assertEquals(zookeeperRequestTimeout, client.getZookeeperClient().getZooKeeper().getClientConfig().getProperty(ZKClientConfig.ZOOKEEPER_REQUEST_TIMEOUT));
+            
+        } catch (NoSuchMethodError e) {
+			log.debug("NoSuchMethodError: ", e);
```

Comment:
Can you please fix formatting here?

Otherwise I will do it as follow up.
I will merge this patch today.

We are also going to cut a release as soon as possible

Reply:
Sure, done.
BTW, at my IDE this formatting, formatting was fine, depends on IDE config, I will notice it.

Reply:
@eolivelli thanks, can be merged and then resolve the Jira issue